LocalStorageの使用

caramelribon opened the pull request:
## LocalStorageの使用

LocalStorageにタスクが保存できるように編集しました

- [x] お手本通りの見た目であるか　ホバー時の挙動、レスポンシブ対応など要確認
- [x] ブランチ名、ファイル名、クラス名、インデントなど、コーディング規則を逸脱していないか
- [x] 記述の重複はないか

GitHubPagesのページのリンク
[リンク](https://caramelribon.github.io/wakana-tatsuta-frontend-basic-2022-todoapp/)

caramelribon committed:
LocalStorageの使用

LocalStorageにタスクが保存できるように編集しました

kodai01 commented on src/components/Organisms/TodoCard/index.jsx:
```diff
@@ -5,58 +5,92 @@ import Task from "../../Molecules/Task/index";
+              <StyledTodoListItem key={index}>
+                <Task
+                  defaultValue={task.name}
+                  defaultIsEditing={initializedAction ? false : true}
```
bool値なので`!initializedAction`と書くだけで動作すると思います

kodai01 commented on src/components/Organisms/TodoCard/index.jsx:
```diff
@@ -5,58 +5,92 @@ import Task from "../../Molecules/Task/index";
-    return task.state === "TODO";
-  });
+  // LocalStorageから取得するデータのKey
+  const localStorageGetKey = JSON.stringify("taskList");
```
`"taskList"`は文字列なのでJSONにする必要ないと思います

kodai01 commented on src/components/Organisms/TodoCard/index.jsx:
```diff
@@ -5,58 +5,92 @@ import Task from "../../Molecules/Task/index";
+  // 初期化時処理(LocalStorageからタスクリストを取得)
+  useEffect(() => {
+    const firstTaskListData = localStorage.getItem(localStorageGetKey);
+    if (firstTaskListData === null) {
+    } else {
```
何も書かないならnullじゃない場合21,22行目を実行するという事にすれば良いと思います

caramelribon committed:
LocalStorageの使用

・LocalStorageのKeyの修正

KeyにしたtaskListは文字列であるので、わざわざJSON文字列に直す必要がないため修正

・初期化処理の修正

初期化時のLocalStorageから受け取ったデータがある場合、taskListに追加するという処理の書き方に修正

・initializedActionの修正

defaultIsEditingの部分で、initializedActionのtrue,falseによる処理を!を使うことによって簡略して書けるため、修正

caramelribon commented:
## LocalStorageの使用2nd

指摘された箇所を修正しました。

- [x] お手本通りの見た目であるか　ホバー時の挙動、レスポンシブ対応など要確認
- [x] ブランチ名、ファイル名、クラス名、インデントなど、コーディング規則を逸脱していないか
- [x] 記述の重複はないか

GitHubPagesのページのリンク
[リンク](https://caramelribon.github.io/wakana-tatsuta-frontend-basic-2022-todoapp/)

kodai01 reviewed (approved):
LocalStorageの機能を上手く活用できていると思います。
